tests/components/flick_electric/test_init.py

```diff
@@ -1,135 +1,154 @@
-            version=1,
-        )
-        entry.add_to_hass(hass)
-
-        # ensure setup fails
-        assert not await hass.config_entries.async_setup(entry.entry_id)
-        assert entry.state is ConfigEntryState.MIGRATION_ERROR
-        await hass.async_block_till_done()
-
-        # Ensure reauth flow is triggered
-        await hass.async_block_till_done()
-        assert len(hass.config_entries.flow.async_progress()) == 1
+    await setup_integration(hass, mock_old_config_entry)
+
+    assert mock_old_config_entry.state is ConfigEntryState.MIGRATION_ERROR
+
+    # Ensure reauth flow is triggered
+    await hass.async_block_till_done()
+    assert len(hass.config_entries.flow.async_progress()) == 1
+
+
+async def test_fetch_fresh_token(
+    hass: HomeAssistant,
+    mock_config_entry: MockConfigEntry,
+    mock_flick_client: AsyncMock,
+) -> None:
+    """Test fetching a fresh token."""
+    await setup_integration(hass, mock_config_entry)
+
+    with patch(
+        "homeassistant.components.flick_electric.config_flow.SimpleFlickAuth.get_new_token",
+        return_value={CONF_ID_TOKEN: NEW_TOKEN},
+    ) as mock_get_new_token:
+        auth = HassFlickAuth(hass, mock_config_entry)
+
+        assert await auth.async_get_access_token() == NEW_TOKEN
+        assert mock_get_new_token.call_count == 1
+
+
+async def test_reuse_token(
+    hass: HomeAssistant,
+    mock_config_entry: MockConfigEntry,
+    mock_flick_client: AsyncMock,
+) -> None:
+    """Test reusing entry token."""
+    await setup_integration(hass, mock_config_entry)
+
+    hass.config_entries.async_update_entry(
+        mock_config_entry,
+        data={
+            **mock_config_entry.data,
+            CONF_ACCESS_TOKEN: {CONF_ID_TOKEN: EXISTING_TOKEN},
+            CONF_TOKEN_EXPIRY: dt_util.now().timestamp() + 3600,
+        },
+    )
+
+    with patch(
+        "homeassistant.components.flick_electric.config_flow.SimpleFlickAuth.get_new_token",
+        return_value={CONF_ID_TOKEN: NEW_TOKEN},
+    ) as mock_get_new_token:
+        auth = HassFlickAuth(hass, mock_config_entry)
+
+        assert await auth.async_get_access_token() == EXISTING_TOKEN
+        assert mock_get_new_token.call_count == 0
+
+
+async def test_fetch_expired_token(
+    hass: HomeAssistant,
+    mock_config_entry: MockConfigEntry,
+    mock_flick_client: AsyncMock,
+) -> None:
+    """Test fetching token when existing token is expired."""
+    await setup_integration(hass, mock_config_entry)
+
+    hass.config_entries.async_update_entry(
+        mock_config_entry,
+        data={
+            **mock_config_entry.data,
+            CONF_ACCESS_TOKEN: {CONF_ID_TOKEN: EXPIRED_TOKEN},
+            CONF_TOKEN_EXPIRY: dt_util.now().timestamp() - 3600,
+        },
+    )
+
+    with patch(
+        "homeassistant.components.flick_electric.config_flow.SimpleFlickAuth.get_new_token",
+        return_value={CONF_ID_TOKEN: NEW_TOKEN},
+    ) as mock_get_new_token:
```

Comment:
You could consider creating a separate mock for the Auth, since you can then just set the value without patch